samples/Azure.ResourceManager.Sample/Generated/Extensions/SubscriptionExtensions.cs

```diff
@@ -0,0 +1,914 @@
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.ResourceManager.Core;
+using Azure.ResourceManager.Core.Resources;
+
+namespace Azure.ResourceManager.Sample
+{
+    /// <summary> Extension methods for convenient access on SubscriptionOperations in a client. </summary>
+    public static partial class SubscriptionExtensions
+    {
+        #region AvailabilitySet
+        private static AvailabilitySetsRestOperations GetAvailabilitySetsRestOperations(ClientDiagnostics clientDiagnostics, TokenCredential credential, ArmClientOptions clientOptions, string subscriptionId, Uri endpoint = null)
+        {
+            var httpPipeline = ManagementPipelineBuilder.Build(credential, endpoint, clientOptions);
+            return new AvailabilitySetsRestOperations(clientDiagnostics, httpPipeline, subscriptionId, endpoint);
+        }
+
+        /// <summary> Lists the AvailabilitySets for this Azure.ResourceManager.Core.SubscriptionOperations. </summary>
+        /// <param name="subscription"> The <see cref="SubscriptionOperations" /> instance the method will execute against. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <return> A collection of resource operations that may take multiple service requests to iterate over. </return>
+        public static AsyncPageable<AvailabilitySet> ListAvailabilitySetAsync(this SubscriptionOperations subscription, CancellationToken cancellationToken = default)
+        {
+            return subscription.ListResources((baseUri, credential, options) =>
+            {
+                var clientDiagnostics = new ClientDiagnostics(options);
+                var restOperations = GetAvailabilitySetsRestOperations(clientDiagnostics, credential, options, subscription.Id.SubscriptionId, baseUri);
+                var result = ListBySubscriptionAsync(clientDiagnostics, restOperations);
+                return new PhWrappingAsyncPageable<AvailabilitySetData, AvailabilitySet>(
+                result,
+                s => new AvailabilitySet(subscription, s));
+            }
+            );
+        }
+
+        /// <summary> Lists all availability sets in a subscription. </summary>
+        /// <param name="clientDiagnostics"> The handler for diagnostic messaging in the client. </param>
+        /// <param name="restOperations"> Resource client operations. </param>
+        /// <param name="expand"> The expand expression to apply to the operation. Allowed values are &apos;instanceView&apos;. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        private static AsyncPageable<AvailabilitySetData> ListBySubscriptionAsync(ClientDiagnostics clientDiagnostics, AvailabilitySetsRestOperations restOperations, string expand = null, CancellationToken cancellationToken = default)
```

Comment:
should probably name this ListAvailabilitySetsBySubscriptionAsync to differenitiate with other methods in the file?

Reply:
I can do `AvailabilitySetsListBySubscriptionAsync`, otherwise I've to break paging method name into `List` and `BySubscription` and then do List + resource name + BySubscription. 

We have to do the same for `ListAll` method.

Since this is a private method which returns different type for each resource makes it different with other methods.